app/services/guidance_link_service.rb

```diff
@@ -0,0 +1,85 @@
+class GuidanceLinkService
+  class << self
+    def call(document:, chapter: nil, just_the_page_number: false)
+      feature_flag = FeatureFlags.enabled?(:mtr_phase_1, without_session_data: true) ? :mtr_phase_1 : :pre_mtr_phase_1
+
+      if just_the_page_number
+        mapping[feature_flag].dig(document, :sections, chapter)
+      elsif chapter
+        mapping[feature_flag].dig(document, :page_url).to_s << "#page=" << mapping[feature_flag].dig(document, :sections, chapter).to_s
```

Comment:
This works, but it's not the most idiomatic way of assembling a string. I'd suggest:

```ruby
"#{mapping[feature_flag].dig(document, :page_url)}#page=#{mapping[feature_flag].dig(document, :sections, chapter)}"
```